### PR TITLE
Brave Today: optimize fetching

### PR DESCRIPTION
**Pull request by petemill**
- Do not fetch any data from remote until user has interacted with brave today once Fix https://github.com/brave/brave-browser/issues/13059
- Fix double-fetching due to feed fetch also fetching sources which notifies feed there is an update which then performs feed fetch... Fix https://github.com/brave/brave-browser/issues/13058
- Debounce fetch feed update when user is toggling publishers

## Submitter Checklist:

- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed).
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_new_tab_ui/async/today.ts [hunk 1]
```diff
@@ -82,12 +82,17 @@ handler.on<number>(Actions.feedItemViewedCountChanged.getType(), async (store, p
...
+  Background.send<{}, Messages.SetPublisherPrefPayload>(MessageTypes.setPublisherPref, {
     publisherId,
     enabled
-  })
-  store.dispatch(Actions.dataReceived({ publishers }))
-  store.dispatch(Actions.checkForUpdate())
+  }).catch((e) => console.error(e))
+  // Refreshing of content after prefs changed is throttled, so wait
+  // a while before seeing if we have new content yet.
+  // This doens't have to be exact since we often check for update when
```
Comment (bsclifton) [inline components/brave_new_tab_ui/async/today.ts:91]: s/doens't/doesn't

**Review by bsclifton — APPROVED** [review]
Tested changes, works good! Code changes look good too. Async startup/shutdown when Today enabled/disabled, debounced re-fetch for publisher pref changes. Very nice 😄

**Review by ryanml — APPROVED** [review]
Works great!

**Comment by petemill** [comment]
CI passed apart from Windows but that was "unstable" not "fail" so 👍

**Comment by petemill** [comment]
Pushed [changes that were only comment updates](https://github.com/brave/brave-core/compare/4aacfd6749337ed9e68808b462e65cdcb43a69e0..d174f90d3a6c6c9fe8e30314a0e2b8d9e77516af)

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 10 x64` using the following build:

```
Brave | 1.20.17 Chromium: 87.0.4280.88 (Official Build) nightly (64-bit)
-- | --
Revision | 89e2380a3e36c3464b5dd1302349b1382549290d-refs/branch-heads/4280@{#1761}
OS | Windows 10 OS Version 2009 (Build 19042.630)
```

Followed the cases/summary motioned via both https://github.com/brave/brave-browser/issues/13059 &  https://github.com/brave/brave-browser/issues/13058 and ran through the following cases:

* ensured that both `https://brave-today-cdn.brave.com` & `https://pcdn.brave.com` are not being contacted when:
 * launching Brave for the first time
 * opening a NTP without interacting/scrolling through Brave Today
* ensured that both `https://brave-today-cdn.brave.com` & `https://pcdn.brave.com` are only initially contacted when:
  * user starts scrolling through Brave Today via a NTP
  * open the NTP `Customize` modal and clicking on `Brave Today` (will load `sources.json` from `https://brave-today-cdn.brave.com`
  * `https://brave-today-cdn.brave.com` --> downloads `feed.json` & `sources.json`
  * `https://pcdn.brave.com` --> pulls down the images
* ensured that when a change is made/publisher is added/removed, Brave contacts `https://brave-today-cdn.brave.com` and pulls a new `feed.json` & `sources.json`
* ensured that scrolling through the Brave Today feed, only `https://pcdn.brave.com` is contacted while pulling down images

**`Scrolling through Brave Today for the first time`**:

**`Example 1`** | **`Example 2`** | 
-----------------|-------------------
BraveTodayInitial1 | BraveTodayInitial2

**`Scrolling through Brave Today feed`**:

**`Example 1`** | **`Example 2`** | 
-----------------|-------------------
TodayScrolling1 | TodayScrolling2

**`Open Brave Today section via Customize for the first time`**:

source

**Pull request merged**